homeassistant/components/freebox/sensor.py

```diff
@@ -118,10 +118,40 @@ async def async_on_demand_update(self):
 
     async def async_added_to_hass(self):
         """Register state update callback."""
-        self._unsub_dispatcher = async_dispatcher_connect(
-            self.hass, self._router.signal_sensor_update, self.async_on_demand_update
+        self.async_on_remove(
+            async_dispatcher_connect(
+                self.hass,
+                self._router.signal_sensor_update,
+                self.async_on_demand_update,
+            )
         )
 
-    async def async_will_remove_from_hass(self):
-        """Clean up after entity before removal."""
-        self._unsub_dispatcher()
+
+class FreeboxCallSensor(FreeboxSensor):
+    """Representation of a Freebox call sensor."""
+
+    def __init__(
+        self, router: FreeboxRouter, sensor_type: str, sensor: Dict[str, any]
+    ) -> None:
+        """Initialize a Freebox call sensor."""
+        self._call_list_for_type = []
+        super().__init__(router, sensor_type, sensor)
+
+    def update(self) -> None:
```

Comment:
Since there is no I/O happening in this update, this can be async.
```suggestion
    async def async_update(self) -> None:
```

But actually, you should just put this inside `async_on_demand_update`. 

Also, you should make `async_on_demand_update` a `@callback` and instead of scheduling an update after writing the latest state, you should just call `self.async_write_ha_state()`

Reply:
Tried this 0fce4c7

But the sensors got `Unknown` state at HA start, after 30s wait it updates the states

Reply:
Is 7174974 ok ?